testgrid/tgapi/pkg/handlers/primary_join.go

```diff
@@ -0,0 +1,88 @@
+package handlers
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/replicatedhq/kurl/testgrid/tgapi/pkg/logger"
+	"github.com/replicatedhq/kurl/testgrid/tgapi/pkg/testinstance"
+)
+
+type JoinPrimaryRequest struct {
+	PrimaryJoin   string `json:"primaryJoin"`
+	SecondaryJoin string `json:"secondaryJoin"`
+}
+
+func AddNodeJoinCommand(w http.ResponseWriter, r *http.Request) {
+	JoinPrimaryRequest := JoinPrimaryRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&JoinPrimaryRequest); err != nil {
+		logger.Error(err)
+		JSON(w, 500, nil)
+		return
+	}
+
+	instanceID := mux.Vars(r)["instanceId"]
+
+	if err := testinstance.AddNodeJoinCommand(instanceID, JoinPrimaryRequest.PrimaryJoin, JoinPrimaryRequest.SecondaryJoin); err != nil {
+		logger.Error(err)
+		JSON(w, 500, nil)
+		return
+	}
+
+	JSON(w, 200, nil)
+}
+
+func GetNodeJoinCommand(w http.ResponseWriter, r *http.Request) {
+	instanceID := mux.Vars(r)["instanceId"]
+
+	primaryJoin, secondaryJoin, err := testinstance.GetNodeJoinCommand(instanceID)
+	if err != nil {
+		logger.Error(err)
+		JSON(w, 500, err)
+		return
+	}
+	if secondaryJoin != "" {
+		data, err := base64.StdEncoding.DecodeString(secondaryJoin)
+		if err != nil {
+			logger.Error(err)
+			JSON(w, 500, err)
+			return
+		}
+		actual := strings.Trim(string(data), "\r\n")
```

Comment:
Not able to do that!
when I run the command from the variable it will have a pipe (|) which is not recognized somehow when I just run the command from the variable, so I have to divide the command into multiple variables and feed the actual command with these variables.